dpctl/tensor/_indexing_functions.py

```diff
@@ -293,7 +293,7 @@ def place(arr, mask, vals):
         raise dpctl.utils.ExecutionPlacementError
     if arr.shape != mask.shape or vals.ndim != 1:
         raise ValueError("Array sizes are not as required")
-    if vals.size == 0:
+    if vals.size == 0 and dpt.nonzero(mask)[0].size:
```

Comment:
The number of set elements in the mask is computed on line 299 and assigned to variable `nz_count`. 

Perhaps it is cleaner to move the check for `vals.size == 0` till after `nz_count` value is known?